internal/controller/nonadminbackup_controller_test.go

```diff
@@ -18,66 +18,636 @@ package controller
+						Namespace: oadpNamespaceName,
+					},
+					Spec: v1.BackupSpec{
+						IncludedNamespaces: []string{nonAdminNamespaceName},
 					},
-					// TODO(user): Specify other spec details if needed.
 				}
-				gomega.Expect(k8sClient.Create(ctx, resource)).To(gomega.Succeed())
+				gomega.Expect(k8sClient.Create(ctx, veleroBackup)).To(gomega.Succeed())
 			}
-		})
-
-		ginkgov2.AfterEach(func() {
-			// TODO(user): Cleanup logic after each test, like removing the resource instance.
-			resource := &nacv1alpha1.NonAdminBackup{}
-			err := k8sClient.Get(ctx, typeNamespacedName, resource)
-			gomega.Expect(err).NotTo(gomega.HaveOccurred())
-
-			ginkgov2.By("Cleanup the specific resource instance NonAdminBackup")
-			gomega.Expect(k8sClient.Delete(ctx, resource)).To(gomega.Succeed())
-		})
-		ginkgov2.It("should successfully reconcile the resource", func() {
-			ginkgov2.By("Reconciling the created resource")
-			controllerReconciler := &NonAdminBackupReconciler{
-				Client: k8sClient,
-				Scheme: k8sClient.Scheme(),
+
+			if scenario.priorStatus != nil {
+				nonAdminBackup.Status = *scenario.priorStatus
+				gomega.Expect(k8sClient.Status().Update(ctx, nonAdminBackup)).To(gomega.Succeed())
 			}
+			// easy hack to test that only one update call happens per reconcile
+			// priorResourceVersion, err := strconv.Atoi(nonAdminBackup.ResourceVersion)
+			// gomega.Expect(err).To(gomega.Not(gomega.HaveOccurred()))
 
-			_, err := controllerReconciler.Reconcile(ctx, reconcile.Request{
-				NamespacedName: typeNamespacedName,
-			})
-			gomega.Expect(err).NotTo(gomega.HaveOccurred())
-			// TODO(user): Add more specific assertions depending on your controller's reconciliation logic.
-			// Example: If you expect a certain status condition after reconciliation, verify it here.
-		})
+			result, err := (&NonAdminBackupReconciler{
+				Client:        k8sClient,
+				Scheme:        testEnv.Scheme,
+				OADPNamespace: oadpNamespaceName,
+			}).Reconcile(
+				context.Background(),
+				reconcile.Request{NamespacedName: types.NamespacedName{
+					Namespace: nonAdminNamespaceName,
+					Name:      testNonAdminBackupName,
+				}},
+			)
+			gomega.Expect(result).To(gomega.Equal(scenario.result))
+			if scenario.resultError == nil {
+				gomega.Expect(err).To(gomega.Not(gomega.HaveOccurred()))
+			} else {
+				gomega.Expect(err).To(gomega.HaveOccurred())
+				gomega.Expect(err.Error()).To(gomega.Equal(scenario.resultError.Error()))
+			}
+
+			gomega.Expect(k8sClient.Get(
+				ctx,
+				types.NamespacedName{
+					Name:      testNonAdminBackupName,
+					Namespace: nonAdminNamespaceName,
+				},
+				nonAdminBackup,
+			)).To(gomega.Succeed())
+
+			gomega.Expect(checkTestNonAdminBackupStatus(nonAdminBackup, scenario.status)).To(gomega.Succeed())
+			if scenario.priorStatus != nil {
+				if len(scenario.priorStatus.VeleroBackupName) > 0 {
+					gomega.Expect(reflect.DeepEqual(
```

Comment:
Is this something incomplete or residual/debug code ? 
We are comparing the NAB.spec.backupSpec with VeleroBackupSpec with just IncludedNS. Let me know if am missing something here.

Reply:
just to confirm NAB spec was updated by reconcile loop